pkg/logging/logging.go

```diff
@@ -0,0 +1,182 @@
+package logging
+
+import (
+	"errors"
+	"flag"
+	"github.com/go-kit/kit/log"
+	"os"
+	"reflect"
+	"runtime"
+)
+
+type FilteredLogger struct {
+	logger                log.Logger
+	component             string
+	filterLevel           int
+	currentLogLevel       int
+	verbosityLevel        int
+	currentVerbosityLevel int
+}
+
+func InitializeLogging(comp string) {
+	// FIXME: change flag to simply be "v" once conflict with glog is resolved
+	flag.IntVar(&defaultVerbosity, "verb", 0, "Verbosity level of logs")
+	flag.StringVar(&defaultComponent, "component", comp, "Default component for logs")
+}
+
+// Wrap a go-kit logger in a FilteredLogger. Not cached
+func MakeLogger(logger log.Logger) *FilteredLogger {
+	defaultLogLevel, ok := logFilterLevels["info"]
+	if !ok {
+		defaultLogLevel = 10
+	}
+	return &FilteredLogger{logger: logger,
+		component:             defaultComponent,
+		filterLevel:           defaultLogLevel,
+		currentLogLevel:       defaultLogLevel,
+		verbosityLevel:        defaultVerbosity,
+		currentVerbosityLevel: defaultVerbosity,
+	}
+}
+
+type NullLogger struct{}
+
+func (n NullLogger) Log(params ...interface{}) error { return nil }
+
+var loggers = make(map[string]*FilteredLogger)
+var initialized = false
+var defaultComponent = ""
+var defaultVerbosity = 0
+
+func Logger(component string) *FilteredLogger {
+	if _, ok := loggers[component]; !ok {
+		logger := log.NewLogfmtLogger(os.Stderr)
+		log := MakeLogger(logger)
+		log.component = component
+		loggers[component] = log
+	}
+	return loggers[component]
+}
+
+func (l *FilteredLogger) SetLogger(logger log.Logger) *FilteredLogger {
+	l.logger = logger
+	return l
+}
+
+var logFilterLevels = map[string]int{
+	"debug":    0,
+	"info":     10,
+	"warning":  20,
+	"error":    30,
+	"critical": 40}
+
+type LogError struct {
+	message string
+}
+
+func (e LogError) Error() string {
+	return e.message
+}
+
+func (l FilteredLogger) Log(params ...interface{}) error {
+	// messages should be logged if any of these conditions are met:
+	// The log filtering level is debug
+	// The log filtering level is info and verbosity checks match
+	// The log message priority is warning or higher
+	force := (l.filterLevel == logFilterLevels["debug"]) || (l.currentLogLevel >= logFilterLevels["warning"])
+
+	if force || (l.filterLevel == logFilterLevels["info"] &&
+		(l.currentLogLevel == l.filterLevel) &&
+		(l.currentVerbosityLevel <= l.verbosityLevel)) {
+		logParams := make([]interface{}, 0)
+		logParams = append(logParams, "component", l.component)
+
+		if l.currentVerbosityLevel > 1 {
+			_, fileName, lineNumber, _ := runtime.Caller(1)
+			logParams = append(logParams, "filename", fileName)
+			logParams = append(logParams, "line", lineNumber)
+		}
+		logParams = append(logParams, params...)
+		return l.logger.Log(logParams...)
+
+	}
+	return nil
+}
+
+func (l FilteredLogger) Object(obj ...interface{}) error {
```

Comment:
Just to highlight this, I would prefer that we only have a convenience builder on top of the existing logging solution from go-kit, not a logging implementation.